src/common/mfu_daos.c

```diff
@@ -1313,3 +1326,2840 @@ int mfu_flist_copy_daos(
+            char buf[buf_len];
+
+            memset(&sgl, 0, sizeof(sgl));
+            memset(&iov, 0, sizeof(iov));
+
+            /* set iod values */
+            (*iod).iod_type  = DAOS_IOD_ARRAY;
+            (*iod).iod_size  = size;
+            (*iod).iod_nr    = 1;
+            (*iod).iod_recxs = &recxs[i];
+
+            /* set sgl values */
+            sgl.sg_nr     = 1;
+            sgl.sg_nr_out = 0;
+            sgl.sg_iovs   = &iov;
+
+            d_iov_set(&iov, buf, buf_len);  
+            /* fetch recx values from source */
+            rc = daos_obj_fetch(*oh, DAOS_TX_NONE, 0, dkey, 1, iod,
+                                &sgl, NULL, NULL);
+            if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to fetch object "DF_RC, DP_RC(rc));
+                goto out;
+            }
+            /* write data to record dset */
+            hdf5->mem_dims[0] = recxs[i].rx_nr;
+            hdf5->rx_memspace = H5Screate_simple(1, hdf5->mem_dims,
+                                                 hdf5->mem_dims);
+            if (hdf5->rx_memspace < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to create rx_memspace");
+                rc = 1;
+                goto out;
+            }
+            /* extend dataset */
+            hdf5->rx_dims[0] += recxs[i].rx_nr;
+            status = H5Dset_extent(hdf5->rx_dset, hdf5->rx_dims);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to extend rx_dset");
+                rc = 1;
+                goto out;
+            }
+            /* retrieve extended dataspace */
+            hdf5->rx_dspace = H5Dget_space(hdf5->rx_dset);
+            if (hdf5->rx_dspace < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to get rx_dspace");
+                rc = 1;
+                goto out;
+            }
+            hsize_t dset_size = H5Sget_simple_extent_npoints(hdf5->rx_dspace);
+            hsize_t start = (hsize_t)recxs[i].rx_idx;
+            hsize_t count = (hsize_t)recxs[i].rx_nr;
+            status = H5Sselect_hyperslab(hdf5->rx_dspace,
+                                         H5S_SELECT_AND, &start,
+                                         NULL, &count, NULL);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to select hyperslab");
+                rc = 1;
+                goto out;
+            }
+
+            hdf5->rx_dtype = H5Tcreate(H5T_OPAQUE, (*iod).iod_size);
+            if (hdf5->rx_dtype < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to create rx_dtype");
+                rc = 1;
+                goto out;
+            }
+            status = H5Dwrite(hdf5->rx_dset, hdf5->rx_dtype,
+                              hdf5->rx_memspace, hdf5->rx_dspace,
+                              H5P_DEFAULT, sgl.sg_iovs[0].iov_buf);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to write rx_dset");
+                rc = 1;
+                goto out;
+            }
```

Comment:
@dsikich For a simple dataset I get:
```
HDF5-DIAG: Error detected in HDF5 (1.13.0) MPI-process 0:
#000: H5D.c line 1156 in H5Dwrite(): can't synchronously write data
...
  #005: H5Dio.c line 395 in H5D__write(): unable to set up type info
    major: Dataset
    minor: Unable to initialize object
  #006: H5Dio.c line 693 in H5D__typeinfo_init(): unable to convert between src and dest datatype
    major: Dataset
    minor: Feature is unsupported
    minor: Write failed
```

Reply:
@daltonbohning hmm I must have updated something that I thought wouldn't matter before I pushed this since this is new. Thanks, I will take a look.

Reply:
@daltonbohning can you try again with my latest push. I deleted the set_tag functions and forgot one, so I think that what causing the write error.

Reply:
@dsikich That seems to have worked, thanks!